runner/status.go

```diff
@@ -69,6 +69,7 @@ type ProcessStatus struct {
 
 	State ProcessState
 	// References to stdout and stderr, not their text
+	// Runner impls shall provide valid refs for all States (but optionally may not for UNKNOWN/BADREQUEST).
```

Comment:
I view it in the same vein as RunId, I want to provide all the ways to address a run at inception.